### DISCUSSION TITLE
2.13in e-Paper hat

### DISCUSSION THREAD

**Discussion by elsardine6k7** [aluminum-ice/pwnagotchi#106] [Q&A]
Ok so I have a pi zero 2 with a2.13in e-paper hat with a gray rainbow sticker with v4 on it. Is the 2.13in display supported?

Installed the latest soft with Etcher (perhaps I Should use Pi loader??) , configured the config.toml like so...

main.name = "Death_Star"
main.lang = "en"
main.whitelist = [
  "EXAMPLE_NETWORK",
  "ANOTHER_EXAMPLE_NETWORK",
  "fo:od:ba:be:fo:od",
  "fo:od:ba"
]

main.plugins.grid.enabled = true
main.plugins.grid.report = true
main.plugins.grid.exclude = [
  "YourHomeNetworkHere"
]

ui.display.enabled = true
ui.display.type = "waveshare_4"
ui.display.color = "black"
......

But I still have no display... Upon boot I get green light from pi after a few mins of flashing, There is a faint flash at the top left hand corner of the display, the green light stops for a minute then starts the cycle.... Not sure if I should use (waveshare213inb_v4)(added in v1.7.7) in place of  or (waveshare_4) (added in v1.8.1 by @marvellousmarvolo )

**Comment by elsardine6k7** [comment]
I got it to come alive but it only stay on for say 5 to 10 min when I use notepad++ to save as .toml and use waveshare_3...  If I just make a txt doc in the root and  save as (config.toml) (all file) it stays as .txt as the type and wont boot.. Am i missing something??   Also i have not seen it connect to my computer, ive used multiple cables but to no luck... 

rspi 02w
tried all the firm down to V1.8.1

**Reply by wpa-2** [reply]
rename it so it has an extension as .toml 
Make sure the spacing is correct on the file or it wont boot. 
```
main.name = "pwnagotchi"
main.lang = "en"
main.whitelist = [
  "EXAMPLE_NETWORK",
  "ANOTHER_EXAMPLE_NETWORK",
  "fo:od:ba:be:fo:od",
  "fo:od:ba"
]

main.plugins.grid.enabled = true
main.plugins.grid.report = true
main.plugins.grid.exclude = [
  "YourHomeNetworkHere"
]

ui.display.enabled = true
ui.display.type = "waveshare_3"
ui.display.color = "black"
ui.fps = 1
```

Notice the spacing if its not correct it wont boot. 
This isnt really a github issue but more of a general linux issue.

**Reply by elsardine6k7** [reply]
Ok i changed some stuff
I did right click add new text doc named it (config.toml) still show up as text document type. ie .txt... wont boot...
If I use notepad++ put the name as (config.toml) it save it as a toml file but removes .toml from the name but it boots up in till.. AI ready where it sits... same results on pi 02w and pi 4b... used pi 4b just to make sure it wasn't the pi 02w... tested with Ethcher and pi loader... same results....
I did check spacing assuming it is now correct due to partial boot..
bought a data micro cable.. no com .. no ssh....

**Reply by elsardine6k7** [reply]
Update I believe the pi 02w board is bad. Put the screen and and files on the pi 4b It boots and currently is sitting at the neural network is ready AI screen. But everything is zero acrosed the board. Ie ch,aps,up time,and pwnd.  Gonna stick with the pi4b until new 02w board comes in.

Can't ssh in, can use micro hdmi and keyboard to get into root.
